### PR TITLE
*: Add a function GenByArgs for Error

### PR DESCRIPTION
**Pull request by hanfei1991**
When the query is "select (1,2) < 3" , the error msg should be "ERROR 1241 (21000): Operand should contain 2 column(s)". The column number should be parameterized, so I add a function to implement it.
@coocood @shenli @zimulala @XuHuaiyu PTAL

### CODE REVIEW

**Comment by zimulala** [comment]
The package of ddl also used the function of `Gen`

**Comment by coocood** [comment]
It's unclear how many args an error should accept.

**Comment by coocood** [comment]
What if we need to add an extra argument in a particular place, or some times we don't want to pass arg?

**Comment by hanfei1991** [comment]
It's better to keep the interface like this. We will completely abandon the "Gen" interface in future.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
Why do some errors have comments some don't comment？

**Comment by hanfei1991** [comment]
@zimulala I have deleted some comments.

**Comment by zimulala** [comment]
Fix CI @hanfei1991 
FAIL	github.com/pingcap/tidb/plan [build failed]

**Comment by hanfei1991** [comment]
@zimulala PTAL

File: ddl/ddl.go [hunk 1]
```diff
@@ -1117,7 +1118,11 @@ func (d *ddl) ModifyColumn(ctx context.Context, ident ast.Ident, spec *ast.Alter
 	colName := spec.Column.Name.Name
 	col := table.FindCol(t.Cols(), colName.L)
 	if col == nil {
-		return infoschema.ErrColumnNotExists.Gen("column %s doesn't exist", colName.O)
+		/*
+			mysql> alter table t modify a int not null;
+			ERROR 1054 (42S22): Unknown column 'a' in 't'
+		*/
```
Comment (zimulala) [inline ddl/ddl.go]: Unify format.

**Comment by hanfei1991** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**